src/libs/portal.ts

```diff
@@ -336,48 +378,28 @@ export async function getFromCalCalendar(db: Db, user: string) {
 /**
  * Retrieves the day rotation for a given date.
  * @param date The date to get the rotation for, defaults to today.
- * @returns The day rotation (integer in [1, 6]).
+ * @returns The day rotation (character A-H).
  */
 export async function getDayRotation(date: Date) {
-	const scheduleDate = new Date(date);
-
-	let body;
-	try {
-		body = await request(dayRotationURL);
-	} catch (e) {
-		throw new Error('There was a problem fetching the day rotation!');
-	}
-
-	const data = ical.parseICS(body);
-
-	// School Portal does not give a 404 if calendar is invalid. Instead, it gives an empty calendar.
-	// Unlike Canvas, the portal is guaranteed to contain some sort of data within a span of a year.
-	if (_.isEmpty(data)) {
-		throw new Error('There was a problem fetching the day rotation!');
-	}
-
-	for (const calEvent of Object.values(data)) {
-		if (typeof calEvent.summary !== 'string') {
-			continue;
-		}
+	return withCalSummary(date, summary => {
+		return /([A-H]) Day/.exec(summary)![1];
+	});
+}
 
-		const start = new Date(calEvent.start!);
-		const end = new Date(calEvent.end || '');
+/**
+ * Return whether date is a late start
+ * @param Date the day to check
+ * @returns boolean of whether it is a late start day
+ */
+export async function isLateStart(date: Date) {
+	// [A - G] because there 7 late start schedules (WHY???)
 
-		const startTime = start.getTime();
-		const endTime = end.getTime();
+	const lateStart =
+		(await withCalSummary(date, summary => {
+			return (/[A-G]9 Day/.exec(summary) ?? []).length > 0;
+		})) ?? false;
```

Comment:
Sorry, I wasn't clear—I meant moving the `await withCalSummary()` call into its own variable so we don't need parentheses around it in the return statement.